cmd/api/src/api/v2/agi.go

```diff
@@ -403,27 +404,52 @@ func parseAGMembersFromNodes(nodes graph.NodeSet, selectors model.AssetGroupSele
 		isCustomMember := false
 		// a member is custom if at least one selector exists for that object ID
 		for _, agSelector := range selectors {
-			if agSelector.Selector == node.Properties.Map[common.ObjectID.String()].(string) {
+			if objectId, ok := node.Properties.Map[common.ObjectID.String()].(string); !ok {
+				log.Warnf("objectid is missing for node %d", node.ID)
+			} else if agSelector.Selector == objectId {
 				isCustomMember = true
 			}
 		}
 
+		var (
+			memberObjectId string
+			memberName     string
+		)
+
+		if objectId, ok := node.Properties.Map[common.ObjectID.String()].(string); ok {
+			memberObjectId = objectId
+		} else {
+			log.Warnf("objectid is missing for node %d", node.ID)
+			memberObjectId = ""
+		}
+
+		if name, ok := node.Properties.Map[common.Name.String()].(string); !ok {
```

Comment:
The conditions appear to be flipped here. You're checking if there was an error and then setting the value that isn't ok, but warning and handling the error case in the case it is ok.